testsuites/listener/shared/client_sg/test_replication.py

```diff
@@ -139,6 +141,98 @@ def test_initial_pull_replication(setup_client_syncgateway_test, continuous):
 @pytest.mark.listener
 @pytest.mark.syncgateway
 @pytest.mark.replication
+@pytest.mark.session
+@pytest.mark.parametrize("num_docs, need_attachments, replication_after_backgroundApp", [
+    (1000, True, False),
+    (10000, False, False),
+    (100000, False, False),
+    (100000, False, True)
+])
+def test_initial_pull_replication_background_apprun(setup_client_syncgateway_test, num_docs, need_attachments,
+                                                    replication_after_backgroundApp):
+    """
+    @summary
+    1. Add specified number of documents to sync-gateway.
+    2. Start continous pull replication to pull the docs from a sync_gateway database.
+    3. While docs are getting replicated , push the app to the background
+    4. Verify if all of the docs got pulled and replication completed when app goes background
+    """
+
+    sg_db = "db"
+    ls_db = "ls_db"
+
+    cluster_config = setup_client_syncgateway_test["cluster_config"]
+    sg_mode = setup_client_syncgateway_test["sg_mode"]
+    ls_url = setup_client_syncgateway_test["ls_url"]
+    sg_one_admin = setup_client_syncgateway_test["sg_admin_url"]
+    sg_one_public = setup_client_syncgateway_test["sg_url"]
+    liteserv = setup_client_syncgateway_test["liteserv"]
+    liteserv_platform = setup_client_syncgateway_test["liteserv_platform"]
+    device_enabled = setup_client_syncgateway_test["device_enabled"]
+
+    sg_config = sync_gateway_config_path_for_mode("listener_tests/listener_tests", sg_mode)
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # No command to push the app to background on device, so avoid test to run on ios device
+    if((liteserv_platform.lower() != "ios" and liteserv_platform.lower() != "android") or
+       (liteserv_platform.lower() == "ios" and device_enabled)):
+        pytest.skip('This test only valid for mobile')
+    log_info("ls_url: {}".format(ls_url))
+    log_info("sg_one_admin: {}".format(sg_one_admin))
+    log_info("sg_one_public: {}".format(sg_one_public))
+
+    client = MobileRestClient()
+    client.create_user(sg_one_admin, sg_db, "seth", password="password", channels=["ABC", "NBC"])
+    session = client.create_session(sg_one_admin, sg_db, "seth")
```

Comment:
same as above

Reply:
looks like I modified at few places and forgot at some places. Modifed to the tests that I have added.